uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/request-ctrl.js

```diff
@@ -177,8 +177,7 @@ define([
           .then(function (balanceChange) {
             if (balanceChange) {
               self.balance.change = balanceChange;
-              // the change is negative so adding it will actually subtract it
-              self.balance.closing = self.balance.opening + self.balance.change.amount;
+              self.calculateClosingBalanceChange();
```

Comment:
calculateClosingBalanceChange() was written in order to calculate closing balance based on opening balance and balance change. The initial thought was to have this function to use it in 2 places:
1) inside the calculateBalanceChange()
2) outside the calculateBalanceChange() when opening balance is changed and no other calculations needed to be done apart from calculating closing balance.

However, I removed calculateClosingBalanceChange() after all. I now think it would be safer to re-calculate all balances if you change the contact, because if we extend the logic in calculateBalanceChange() in the future there is a risk of having a similar bug again.